src/client/gui/lib/vm_details/terminal_tabs.dart

```diff
@@ -144,7 +148,32 @@ class TerminalTabs extends ConsumerWidget {
           title: 'Shell ${shellId.id}',
           selected: index == currentIndex,
           onTap: () => ref.read(notifier).setCurrent(index),
-          onClose: () => ref.read(notifier).remove(index),
+          onClose: () {
```

Comment:
Should we ask also when closing the entire GUI? I know we already ask about stopping instances, so perhaps there's a way we could fold this question in there somehow... Something like changing the text of that last popup to also warn about open terminals and offer a cancel button?

And what about "stopping" an instance? Should we ask there too?

Reply:
Hmm, when closing the entire GUI, what should we do with the 'Do not ask again' switch? Show it under the 'Keep instances running' option? And how would a user indicate that they want to close their terminals, but keep the instances running in the background? Should we show the popups one after the other?

About having the popup when you want to stop an instance, yes, I think we can have that.

Reply:
Hmm, but do you think it makes sense to ask to close each terminal but not when closing all terminals? It feels a bit like double standards to me.

Perhaps we could add just a notice that there were running terminals that would be closed. The question could remain about stopping instances, such that the "do not ask again" kept referring to it. 

Perhaps something like [this](https://excalidraw.com/#json=H2sfIpcV4kUQDbJgoK5FG,sQVAlwJ-jRh0gTngfbI1hA): 
![image](https://github.com/user-attachments/assets/7d8c1a1a-8a43-498c-a14f-f0f0ceff7e7a)

The text could be improved. Suggestions welcome, of course.

And the cancel button is actually already there in the form of a cross... I wonder if it should be a proper button too, regardless of terminals, but that is a separate matter.

Reply:
For the record, as discussed elsewhere...

The proposal above would still have the issue that, if the user selected "remember this setting" here but "always ask" on the terminal question, terminals would still be closed with no question or indication when closing the GUI.

As for asking when stopping, we'd need to decide what to do when batch closing.